File: lib/data/remote/model/author_dto.dart
import 'package:json_annotation/json_annotation.dart';

part '../../../generated/data/remote/model/author_dto.g.dart';

@JsonSerializable(fieldRename: FieldRename.snake, createToJson: false)
class AuthorDTO {
  final int id;
  final String name;
  final String? biography;
  final int? sermonsCount;

  AuthorDTO({
    required this.id,
    required this.name,
    required this.biography,
    required this.sermonsCount,
  });

  static const fromJsonFactory = _$AuthorDTOFromJson;

  @override
  String toString() {
    return 'AuthorDTO{id: $id, name: $name, biography: $biography, sermonsCount: $sermonsCount}';
  }
}

File: data/src/main/java/by/geth/gethsemane/data/model/remote/AuthorDTO.kt
package by.geth.gethsemane.data.model.remote

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class AuthorDTO(
    @SerialName("id")
    val id: Long,
    @SerialName("name")
    val name: String,
    @SerialName("biography")
    val biography: String?,
)
